src/js/spatial-navigation.js

```diff
@@ -0,0 +1,496 @@
+   * This method resets the pause flag, re-enabling the navigation logic.
+   */
+  resume() {
+    this.isPaused = false;
+  }
+
+  /**
+   * Handles Player Blur.
+   *
+   */
+  handlePlayerBlur(component) {
+    if (component.name() === 'CloseButton') {
+      this.refocusComponent();
+    } else {
+      this.pause();
+
+      if (component && component.el_) {
+        this.lastFocusedComponent = component;
+      }
+    }
+  }
+
+  /**
+   * Handles Player Focus.
+   *
+   */
+  handlePlayerFocus() {
+    this.resume();
+  }
+
+  /**
+   * Gets a set of focusable components.
+   *
+   * @return {Array}
+   *         Returns an array of focusable components.
+   */
+  getComponents() {
+    const player = this.player;
+    const focusableComponents = [];
+
+    /**
+   * Searchs for children candidates.
+   *
+   * Pushes Components to array of 'focusableComponents'.
+   * Calls itself if there is children elements inside of iterated component.
+   */
+    function searchForChildrenCandidates(componentsArray) {
+      for (const i of componentsArray) {
+        if (i.hasOwnProperty('el_') && i.getIsFocusable(i.el_) && i.getIsAvailableToBeFocused(i.el_)) {
+          focusableComponents.push(i);
+        }
+        if (i.hasOwnProperty('children_') && i.children_.length > 0) {
+          searchForChildrenCandidates(i.children_);
+        }
+      }
+    }
+
+    // Iterate inside of all children components of the player.
+    player.children_.forEach((value) => {
+      if (value.hasOwnProperty('el_')) {
+        // If component has required functions 'getIsFocusable' & 'getIsAvailableToBeFocused', is focusable & avilable to be focused.
+        if (value.getIsFocusable && value.getIsAvailableToBeFocused && value.getIsFocusable(value.el_) && value.getIsAvailableToBeFocused(value.el_)) {
+          focusableComponents.push(value);
+          return;
+          // If component has posible children components as candidates.
+        } else if (value.hasOwnProperty('children_') && value.children_.length > 0) {
+          searchForChildrenCandidates(value.children_);
+          // If component has posible item components as candidates.
+        } else if (value.hasOwnProperty('items') && value.items.length > 0) {
+          searchForChildrenCandidates(value.items);
+          // If there is a suitable child element within the component's DOM element.
+        } else if (this.findSuitableDOMChild(value)) {
+          focusableComponents.push(value);
+        }
+      }
+    });
+
+    this.focusableComponents = focusableComponents;
+    return this.focusableComponents;
+  }
+
+  /**
+   * Finds a suitable child element within the provided component's DOM element.
+   *
+   * @param {Object} component - The component containing the DOM element to search within.
+   * @return {HTMLElement|null} Returns the suitable child element if found, or null if not found.
+   */
+  findSuitableDOMChild(component) {
+    function searchForSuitableChild(node) {
```

Comment:
nit: It seems a little unecessary to create an extra function here, you could just do:

```
const node = component.el()

if (component.getIsFocusable(node) && component.getIsAvailableToBeFocused(node)) {
        return node;
}

...

```